src/Tinker.jl

```diff
@@ -49,12 +66,28 @@ function init_gui(image::AbstractArray; name="Tinker")
         end
     end
 
-    # rectangle draw function
-    function drawrect(ctx, rect, color)
-        set_source(ctx, color)
-        rectangle(ctx, rect[1], rect[2], rect[3], rect[4])
-        stroke(ctx)
-    end;
+    # holds x zoom level
+    xzoom = map(zr) do r
+        100*r.fullview.x.right/(r.currentview.x.right-(r.currentview.x.left-1))
+    end
+
+    # hold y zoom level
+    yzoom = map(zr) do r
+        100*r.fullview.y.right/(r.currentview.y.right-(r.currentview.y.left-1))
+    end
+
+    # performs proportional zoom in (x2 each time)
+    function zoomin()
+        push!(zr, zoom(value(zr), 0.5))
+        println("zoom x%: ", value(xzoom),"\n zoom y%: ", value(yzoom))
+        # why doesn't this update after push?
+    end
+
+    # performs proportional zoom out (x0.5 each time)
+    function zoomout() # doesn't perfectly undo zoomin: problem
```

Comment:
There seems to be some kind of rounding problem with GtkReactive.zoom. When I'm calculating zoom %, a push of 0.5 to the region will zoom the view in to an even 200%; however, pushing 2 to the region will zoom out to something like 100.13037...%.